packages/mui-material/src/Popover/Popover.js

```diff
@@ -244,13 +246,17 @@ const Popover = React.forwardRef(function Popover(inProps, ref) {
       const widthThreshold = containerWindow.innerWidth - marginThreshold;
 
       // Check if the vertical axis needs shifting
-      if (top < marginThreshold) {
+      if (!disableMarginThreshold && top < marginThreshold) {
```

Comment:
As a side note, it's a bit of a mystery to me why we don't use Popper for these calculations.

Reply:
The idea here is to provide a way for allowing the select to scroll without introducing breaking changes in v5.

After this is closed, I plan to open an RFC regarding how to approach the Select and Autocomplete scrolling behavior in v6. We will use Base UI for that so I don't think it's worth it to do a big refactor here.

Reply:
Sure, this was not a remark to this PR but a general one.